streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -368,50 +378,134 @@ void handleLostAll() {
     }
 
     /**
+     * Compute the offset total summed across all stores in a task. Includes offset sum for any tasks we own the
+     * lock for, which includes assigned and unassigned tasks we locked in {@link #tryLockForAllTaskDirectories()}
+     *
      * @return Map from task id to its total offset summed across all state stores
      */
     public Map<TaskId, Long> getTaskOffsetSums() {
         final Map<TaskId, Long> taskOffsetSums = new HashMap<>();
 
-        for (final TaskId id : tasksOnLocalStorage()) {
-            if (isRunning(id)) {
-                taskOffsetSums.put(id, Task.LATEST_OFFSET);
+        for (final TaskId id : lockedTaskDirectories) {
+            final Task task = tasks.get(id);
+            if (task != null) {
+                if (task.isActive() && task.state() == RUNNING) {
+                    taskOffsetSums.put(id, Task.LATEST_OFFSET);
+                } else {
+                    taskOffsetSums.put(id, sumOfChangelogOffsets(task.changelogOffsets()));
+                }
             } else {
-                taskOffsetSums.put(id, 0L);
+                final File checkpointFile = stateDirectory.checkpointFileFor(id);
+                try {
+                    // If we can't read the checkpoint file or it doesn't exist, release the task directory
+                    // so the background cleaner thread can do its thing
+                    if (checkpointFile.exists()) {
+                        taskOffsetSums.put(id, sumOfChangelogOffsets(new OffsetCheckpoint(checkpointFile).read()));
+                    } else {
+                        releaseTaskDirLock(id);
```

Comment:
How about just a warning in `releaseTaskDirLock`, and then log as an error if it's actually fatal?

Reply:
I would prefer to not log anything in `releaseTaskDirLock()` and check the return value of `releaseTaskDirLock()` at caller side. That would avoid double log messages due to the same event.
On a different note, do we need to return the exception from `releaseTaskDirLock()`. We could just throw it and catch it where we call `releaseTaskDirLock()`. Am I missing something?

Reply:
We have/are making some changes to the directory cleanup (unrelated to 441) that seem like we could end up with a lot of empty directories that have to wait for the cleanup thread to be removed. Since every thread has to go through every task directory and try to lock it, I was thinking we should try to avoid blocking the cleanup thread as much as possible.

Note that currently, the cleanup thread runs every 10 min (by default). If we also choose 10 min as the default for the probing rebalance interval, and leave empty directories locked during a rebalance, we might never delete them (until the probing reblances end of course).

Maybe a better approach is to let the cleanup thread run slightly more frequently to remove empty directories only, and skip anything with remaining state & valid checkpoint. WDYT? If that sounds preferable I can make a ticket to follow up later

Reply:
Nevermind the above -- assuming we get [this PR](https://github.com/apache/kafka/pull/8267) into 2.6 as well we can just leverage the new `listNonEmptyTaskDirectories` and the problem becomes moot. Removed the unlocking attempt from this method for now